android-components/components/feature/search/src/test/java/mozilla/components/feature/search/region/RegionMiddlewareTest.kt

```diff
@@ -120,4 +121,28 @@ class RegionMiddlewareTest {
         assertEquals("DE", store.state.search.region!!.home)
         assertEquals("DE", store.state.search.region!!.current)
     }
+
+    @Test
+    fun `GIVEN a locale is already selected WHEN the locale changes THEN update region on RefreshSearchEngines`() = runTestOnMain {
+        val middleware = RegionMiddleware(FakeContext(), locationService, dispatcher)
+        middleware.regionManager = regionManager
+        locationService.region = LocationService.Region("FR", "France")
+        val store = BrowserStore(
+            middleware = listOf(middleware),
+        )
+        store.dispatch(InitAction).joinBlocking()
+        middleware.updateJob?.joinBlocking()
+        store.waitUntilIdle()
+        assertEquals("FR", store.state.search.region!!.home)
+        assertEquals("FR", store.state.search.region!!.current)
+
+        locationService.region = LocationService.Region("DE", "Germany")
+        regionManager.update()
+        store.dispatch(RefreshSearchEnginesAction).joinBlocking()
+        middleware.updateJob?.joinBlocking()
+        store.waitUntilIdle()
+
+        assertEquals("FR", store.state.search.region!!.home)
+        assertEquals("DE", store.state.search.region!!.current)
```

Comment:
```suggestion
        val middleware = RegionMiddleware(FakeContext(), locationService, dispatcher)
        middleware.regionManager = regionManager
        
        locationService.region = LocationService.Region("FR", "France")
        
        val store = BrowserStore(
            middleware = listOf(middleware),
        )
        
        store.dispatch(InitAction).joinBlocking()
        middleware.updateJob?.joinBlocking()
        store.waitUntilIdle()
        
        assertEquals("FR", store.state.search.region!!.home)
        assertEquals("FR", store.state.search.region!!.current)

        locationService.region = LocationService.Region("DE", "Germany")
        regionManager.update()
        
        store.dispatch(RefreshSearchEnginesAction).joinBlocking()
        middleware.updateJob?.joinBlocking()
        store.waitUntilIdle()

        assertEquals("FR", store.state.search.region!!.home)
        assertEquals("DE", store.state.search.region!!.current)
```
Let's add some new lines to help space out the logic and operations so that it's easier to read. There is a sense of grouping here where we space out the variable assignments, dispatch and update calls and the assertions. I mostly followed how the operations were spaced out from the test above. I think what we want to avoid is having long lines of code all squashed together since that makes it harder to read and comprehend the code.

Reply:
I would argue that having clean separations between the 3 main structures of a test: arrange / act / assert that would also match the given / when / then test declaration helps to follow and understand the test more easily - the setup part / the action / what we want to test.